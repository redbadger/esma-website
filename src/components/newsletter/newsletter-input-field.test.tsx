import React from "react";
import renderer from "react-test-renderer";
import NewsletterInputField from "./newsletter-input-field";

describe("NewsletterInputField", () => {
  it("renders correctly with no error state", () => {
    const tree = renderer
      .create(
        <NewsletterInputField
          name="test"
          type="text"
          label="testLabel"
          error={false}
          errorMessage="ERROR!"
          errorSetter={() => {}}
          valueSetter={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly with an error state", () => {
    const tree = renderer
      .create(
        <NewsletterInputField
          name="test"
          type="text"
          label="testLabel"
          error={true}
          errorMessage="ERROR!"
          errorSetter={() => {}}
          valueSetter={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
