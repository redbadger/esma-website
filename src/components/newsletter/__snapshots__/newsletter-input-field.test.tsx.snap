// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsletterInputField renders correctly with an error state 1`] = `
<section
  className="css-usoz6a-newsletterInputFieldStyles"
>
  <label
    htmlFor="testInput"
  >
    testLabel
  </label>
  <input
    className="error"
    id="testInput"
    name="test"
    onChange={[Function]}
    type="text"
  />
  <p
    className="input-error"
  >
    ERROR!
  </p>
</section>
`;

exports[`NewsletterInputField renders correctly with no error state 1`] = `
<section
  className="css-usoz6a-newsletterInputFieldStyles"
>
  <label
    htmlFor="testInput"
  >
    testLabel
  </label>
  <input
    className=""
    id="testInput"
    name="test"
    onChange={[Function]}
    type="text"
  />
  <p
    className="input-error"
  >
    ERROR!
  </p>
</section>
`;
